rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -798,10 +798,85 @@ void getFederationElection_whenElectionIsNotInStorage_shouldReturnDefault() {
         // Assert
 
         byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
+
         assertArrayEquals(expectedElectionEncoded, serializeElection(actualElection));
 
     }
 
+    @Test
+    void getActiveFederationCreationBlockHeight_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(false);
+
+        // Arrange
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        // Putting some value in the storage just to then assert that before fork, the storage won't be accessed.
+        storageAccessor.saveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 1 });
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Assert
+        assertEquals(Optional.empty(), federationStorageProvider.getActiveFederationCreationBlockHeight(activations));
+
+    }
+
+    @Test
+    void getActiveFederationCreationBlockHeight_afterRSKIP186_getsValueFromStorage() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        // Arrange
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        long expectedValue = 1;
+        storageAccessor.saveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { (byte) expectedValue });
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+        Optional<Long> actualValueOptional = federationStorageProvider.getActiveFederationCreationBlockHeight(activations);
+
+        // Assert
+
+        assertTrue(actualValueOptional.isPresent());
+        assertEquals(expectedValue, actualValueOptional.get());
+
+        // Setting in storage a different value to assert that calling the method again should return cached value
+
+        storageAccessor.saveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 2 });
```

Comment:
maybe move the `new byte[] { 2 }` to a variable?

Reply:
No need. It's only used there.